### PR TITLE
add 7 system variables introduced in 6.3 but have no user doc

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
It is found that 7 system variables are missing in [system variables](/system-variables.md). Since all system variables introduced in the product should be documented, this PR is created to document missing variables.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): https://github.com/pingcap/docs/pull/9886, https://github.com/pingcap/docs/pull/10564

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1006,6 +1036,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     * 使用 `SELECT` 读取 `noop` 的系统变量时会报 `"variable *variable_name* has no effect in TiDB"` 的警告。
 - 你可以通过 `SELECT * FROM INFORMATION_SCHEMA.CLIENT_ERRORS_SUMMARY_GLOBAL;` 语句来检查 TiDB 实例是否曾设置和读取 `noop` 系统变量。
 
+### `tidb_enable_null_aware_anti_join` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：
```
Comment (AilinKid) [inline system-variables.md:1041]:
```suggestion
- 作用域：SESSION ｜GLOBAL
```

File: system-variables.md [hunk 2]
```diff
@@ -1006,6 +1036,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 是否持久化到集群：
+- 默认值：
```
Comment (AilinKid) [inline system-variables.md:1043]:
```suggestion
- 默认值：`OFF`
```

File: system-variables.md [hunk 3]
```diff
@@ -1006,6 +1036,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 这个变量用于 XXX。
```
Comment (AilinKid) [inline system-variables.md:1044]:
```suggestion
- 自 v6.3.0 起，TiDB 的对特殊集合算子 `NOT IN` 和 `!= ALL` 引导子查询而产生的 ANTI JOIN 将可以使用此开关控制是否采用 Null Aware Hash Join 的执行方式。该开关默认为 `OFF`。
```

File: system-variables.md [hunk 4]
```diff
@@ -687,6 +695,14 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 范围：`[0, 9223372036854775807]`
 - 这个变量用来控制 DDL 操作失败重试的次数。失败重试次数超过该参数的值后，会取消出错的 DDL 操作。
 
+### `tidb_ddl_flashback_concurrency` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：
+- 是否持久化到集群：
+- 默认值：
+- 范围：
+- 这个变量用来 XXX。
```
Comment (Defined2014) [inline system-variables.md:704]:
```suggestion
- 这个变量用来控制 `flashback cluster` 的并发数。
```

File: system-variables.md [hunk 5]
```diff
@@ -1345,6 +1389,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
     - `txn_mode`：事务模式。可选值：`OPTIMISTIC`（乐观事务模式），或 `PESSIMISTIC`（悲观事务模式）
     - `sql`：当前查询对应的 SQL 语句
 
+### `tidb_general_plan_cache_size` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1392]: @fzzf678

File: system-variables.md [hunk 6]
```diff
@@ -345,6 +345,14 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 默认值：`ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION`
 - 这个变量控制许多 MySQL 兼容行为。详情见 [SQL 模式](/sql-mode.md)。
 
+### `sql_require_primary_key` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:348]: @joycse06

File: system-variables.md [hunk 7]
```diff
@@ -1006,6 +1036,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     * 使用 `SELECT` 读取 `noop` 的系统变量时会报 `"variable *variable_name* has no effect in TiDB"` 的警告。
 - 你可以通过 `SELECT * FROM INFORMATION_SCHEMA.CLIENT_ERRORS_SUMMARY_GLOBAL;` 语句来检查 TiDB 实例是否曾设置和读取 `noop` 系统变量。
 
+### `tidb_enable_null_aware_anti_join` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1039]: @AilinKid

File: system-variables.md [hunk 8]
```diff
@@ -931,6 +947,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 这个变量用来控制是否启用统计信息快速分析功能。默认值 0 表示不开启。
 - 快速分析功能开启后，TiDB 会随机采样约 10000 行的数据来构建统计信息。因此在数据分布不均匀或者数据量比较少的情况下，统计信息的准确度会比较低。这可能导致执行计划不优，比如选错索引。如果可以接受普通 `ANALYZE` 语句的执行时间，则推荐关闭快速分析功能。
 
+### `tidb_enable_foreign_key` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:950]: @crazycs520

File: system-variables.md [hunk 9]
```diff
@@ -1119,6 +1156,13 @@ Query OK, 0 rows affected (0.09 sec)
 - 默认值：`ON`
 - 这个变量用于动态地控制 TiDB 遥测功能是否开启。设置为 `OFF` 可以关闭 TiDB 遥测功能。当所有 TiDB 实例都设置 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-从-v402-版本开始引入) 为 `false` 时将忽略该系统变量并总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)了解该功能详情。
 
+### `tidb_enable_tiflash_read_for_write_stmt` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1159]: @gengliqi

File: system-variables.md [hunk 10]
```diff
@@ -931,6 +947,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
 - 快速分析功能开启后，TiDB 会随机采样约 10000 行的数据来构建统计信息。因此在数据分布不均匀或者数据量比较少的情况下，统计信息的准确度会比较低。这可能导致执行计划不优，比如选错索引。如果可以接受普通 `ANALYZE` 语句的执行时间，则推荐关闭快速分析功能。
 
+### `tidb_enable_foreign_key` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：
+- 是否持久化到集群：
+- 默认值：
+- 这个变量用于 XXX。
+
+### `tidb_enable_general_plan_cache` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:957]: @fzzf678

File: system-variables.md (hunk 7 above)
Comment (AilinKid) [reply system-variables.md:1039]: LGTM

File: system-variables.md [hunk 11]
```diff
@@ -1119,6 +1156,13 @@ Query OK, 0 rows affected (0.09 sec)
 - 默认值：`ON`
 - 这个变量用于动态地控制 TiDB 遥测功能是否开启。设置为 `OFF` 可以关闭 TiDB 遥测功能。当所有 TiDB 实例都设置 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-从-v402-版本开始引入) 为 `false` 时将忽略该系统变量并总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)了解该功能详情。
 
+### `tidb_enable_tiflash_read_for_write_stmt` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：
+- 是否持久化到集群：
+- 默认值：
+- 这个变量用于 XXX。
```
Comment (gengliqi) [inline system-variables.md:1164]:
```suggestion
- 作用域：SESSION | GLOBAL
- 是否持久化到集群：是
- 默认值：`OFF`
- 这个变量用于控制写 SQL 中的读取是否会下推 TiFlash。**需要注意该功能目前仅供测试，请勿在正式环境中使用，该功能正式 release 之后该参数不会保留**
```

File: system-variables.md [hunk 12]
```diff
@@ -1119,6 +1156,13 @@ Query OK, 0 rows affected (0.09 sec)
 - 默认值：`ON`
 - 这个变量用于动态地控制 TiDB 遥测功能是否开启。设置为 `OFF` 可以关闭 TiDB 遥测功能。当所有 TiDB 实例都设置 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-从-v402-版本开始引入) 为 `false` 时将忽略该系统变量并总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)了解该功能详情。
 
+### `tidb_enable_tiflash_read_for_write_stmt` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 这个变量用于控制写 SQL 中的读取是否会下推 TiFlash。**需要注意该功能目前仅供测试，请勿在正式环境中使用，该功能正式 release 之后该参数不会保留**
```
Comment (gengliqi) [inline system-variables.md:1164]:
```suggestion
- 这个变量用于控制写 SQL 中的读取是否会下推 TiFlash。**需要注意该功能目前仅供测试，普通用户请勿更改，该功能正式 release 之后该参数不会保留**
```

Comment (shichun-0415) [reply system-variables.md:1164]: `普通用户` 感觉不太妥。另外，这个参数`以后不会保留`这个信息似乎不用在这里呈现，等到需要移除的时候直接来移除吧。
这么改？
> 这个变量用于控制写 SQL 中的读取是否会下推到 TiFlash。注意，该功能目前仅供测试，请保留其默认值 `OFF`。

Comment (gengliqi) [reply system-variables.md:1164]: 可以的

File: system-variables.md [hunk 13]
```diff
@@ -931,6 +947,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 这个变量用来控制是否启用统计信息快速分析功能。默认值 0 表示不开启。
 - 快速分析功能开启后，TiDB 会随机采样约 10000 行的数据来构建统计信息。因此在数据分布不均匀或者数据量比较少的情况下，统计信息的准确度会比较低。这可能导致执行计划不优，比如选错索引。如果可以接受普通 `ANALYZE` 语句的执行时间，则推荐关闭快速分析功能。
 
+### `tidb_enable_foreign_key` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：
+- 是否持久化到集群：
+- 默认值：
+- 这个变量用于 XXX。
```
Comment (crazycs520) [inline system-variables.md:955]:
```suggestion
### `tidb_enable_foreign_key` <span class="version-mark">从 v6.3.0 版本开始引入</span>

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`OFF`
- 这个变量用于控制是否开启 `FOREIGN KEY` 特性，目前 `FOREIGN KEY` 特性还未 GA，不推荐开启该变量。
```

File: system-variables.md [hunk 14]
```diff
@@ -1024,6 +1068,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     * 使用 `SELECT` 读取 `noop` 的系统变量时会报 `"variable *variable_name* has no effect in TiDB"` 的警告。
 - 你可以通过 `SELECT * FROM INFORMATION_SCHEMA.CLIENT_ERRORS_SUMMARY_GLOBAL;` 语句来检查 TiDB 实例是否曾设置和读取 `noop` 系统变量。
 
+### `tidb_enable_null_aware_anti_join` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION ｜GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 自 v6.3.0 起，TiDB 的对特殊集合算子 `NOT IN` 和 `!= ALL` 引导子查询而产生的 ANTI JOIN 将可以使用此开关控制是否采用 Null Aware Hash Join 的执行方式。该开关默认为 `OFF`。
```
Comment (qiancai) [inline system-variables.md:1076]:
```suggestion
- 这个变量用于控制 TiDB 对特殊集合算子 `NOT IN` 和 `!= ALL` 引导的子查询产生的 ANTI JOIN 是否采用 Null Aware Hash Join 的执行方式。
```

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**